src/main/java/seedu/address/logic/commands/DriveCommand.java

```diff
@@ -0,0 +1,116 @@
+
+import java.util.List;
+import java.util.Set;
+import java.util.StringJoiner;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.human.Name;
+import seedu.address.model.human.Phone;
+import seedu.address.model.human.driver.Driver;
+import seedu.address.model.human.person.Address;
+import seedu.address.model.human.person.Person;
+import seedu.address.model.human.person.TripDay;
+import seedu.address.model.human.person.TripTime;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Associates a Driver with the selected Persons
+ */
+public class DriveCommand extends Command {
+    public static final String COMMAND_WORD = "drive";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Associates a Driver with the selected Commuters. "
+            + "Parameters: "
+            + PREFIX_NAME + "DRIVER NAME "
+            + PREFIX_PHONE + "DRIVER PHONE "
+            + PREFIX_COMMUTER + "COMMUTER "
+            + "[" + PREFIX_COMMUTER + "COMMUTER]...\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NAME + "John Doe "
+            + PREFIX_PHONE + "98765432 "
+            + PREFIX_COMMUTER + "1 "
+            + PREFIX_COMMUTER + "4 ";
+
+    public static final String MESSAGE_NO_COMMUTERS = "No commuters were selected.";
+    public static final String MESSAGE_DRIVE_SUCCESS = "Assigned %s to: %s";
+
+    private final Driver driver;
+    private final Set<Index> persons;
+
+    /**
+     * Creates an AddCommand to add the specified {@code Person}
+     */
+    public DriveCommand(Driver driver, Set<Index> persons) {
+        requireNonNull(driver);
+        requireNonNull(persons);
+        this.driver = driver;
+        this.persons = persons;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        if (persons.size() == 0) {
+            throw new CommandException(MESSAGE_NO_COMMUTERS);
+        }
+        StringJoiner joiner = new StringJoiner(", ");
+
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        for (Index idx : persons) {
+            if (idx.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+        }
+
+        for (Index idx : persons) {
+            Person personToEdit = lastShownList.get(idx.getZeroBased());
+            Person editedPerson = assignDriverToPerson(personToEdit, driver);
+            joiner.add(editedPerson.getName().toString());
+            model.setPerson(personToEdit, editedPerson);
+        }
+
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+
+        return new CommandResult(String.format(MESSAGE_DRIVE_SUCCESS, driver.toString(), joiner.toString()));
+    }
```

Comment:
perhaps this method may seem a little too long? maybe we could use a stream here